guardrails-ai/guardrails
# add support for specifying a template in cli create

Adds support for templates via the guardrails create cli. To test a local file create a template like chatbot.json
```
{
    "name": "chatbot",
    "description": "a basic guard to facilitate chat w llms",
    "template_version": "0.0.1",
    "namespace": "guardrails",
    "guards": [
      {
        id: "chatbot",
        "name": "chatbot",
        "validators": [
            {
                "id": "guardrails/competitor_check",
                "on": "$",
                "onFail": "exception",
                "kwargs": {
                    "competitors": [
                        "OpenAI",
                        "Anthropic"
                    ]
                }
            },
            {
                "id": "guardrails/detect_pii",
                "on": "$",
                "onFail": "exception",
                "kwargs": {
                    "pii_entities": [
                        "EMAIL_ADDRESS",
                        "PHONE_NUMBER"
                    ]
                }
            },
            {
                "id": "guardrails/profanity_free",
                "on": "$",
                "onFail": "exception",
                "kwargs": {}
            }
        ],
      }
    ]
  }
```

run `guardrails create --template template.json`
add a openAI api key like `os.environ["OPENAI_API_KEY"] = "sk-proj"`
run `guardrails start --config config.py`

to test full end to end
run https://github.com/guardrails-ai/validator-hub-ui/pull/22 via `bun run dev`
go to http://localhost:3000/guard-templates
observe
<img width="1255" alt="Screenshot 2024-07-31 at 1 47 21 PM" src="https://github.com/user-attachments/assets/99b3f233-9800-44c8-ad5f-1068e3430b3b">
run `guardrails create --template hub:template://guardrails/chatbot`
run `guardrails start --config config.py --port 8080`

use this script to test both cases
```
import os
import json

os.environ["OPENAI_LOG"] = "debug"

import openai
openai.base_url = "http://localhost:8080/guards/chatbot/openai/v1/"
openai.default_headers = {"x-foo": "true"}
os.environ["OPENAI_API_KEY"] = "sk-proj"
messages = [
        {
            "role": "user",
            "content": "Tell me about the llm llama"
            # "content": "Tell me about chat gpt from openai"
        },
    ]


completion = openai.chat.completions.create(
    model="gpt-4o",
    messages=messages,
)

print("===completion===", completion)
```
